.github/workflows/mypy.yaml

```diff
@@ -0,0 +1,51 @@
+name: "mypy"
+on:
+  push:
+    branches:
+      - develop
+  pull_request:
+    branches:
+      - develop
+
+concurrency:
+
+  group: "${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}"
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  mypy :
+    if: "github.repository == 'MDAnalysis/mdanalysis'"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: setup_miniconda
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          python-version: 3.8
+          auto-update-conda: true
+          channel-priority: flexible
+          channels: conda-forge, bioconda
+          add-pip-as-python-dependency: true
+          mamba-version: "*"
+          architecture: x64
+
+      - name: install_deps
+        uses: ./.github/actions/setup-deps
+        with:
+          mamba: true
+          full-deps: true
+          numpy: numpy=1.20.0
+
+      - name: install
+        run: |
+          python -m pip install mypy
```

Comment:
Do you not need to install MDAnalysis?

Reply:
Umm as we already installed the dependencies should i add 
          cd package
          python setup.py develop --no-deps
instead of 
cd package
python setup.py install ?